src/plugins/intel_gpu/include/intel_gpu/primitives/primitive.hpp

```diff
@@ -130,7 +130,14 @@ struct primitive {
           output_paddings(output_paddings),
           output_data_types(output_data_types),
           input(input),
-          num_outputs(num_outputs) {}
+          num_outputs(num_outputs) {
+            if (output_paddings.size() < num_outputs) {
+                this->output_paddings.insert(this->output_paddings.end(), num_outputs - output_paddings.size(), padding());
+            }
+            if (output_data_types.size() < num_outputs) {
+                this->output_data_types.insert(this->output_data_types.end(), num_outputs - output_data_types.size(), optional_data_type());
+            }
```

Comment:
Wouldn't it be better from design perspective throw an error in such cases?
Instead, we could have empty vectors as default values and fill them in this constructor.
Something like:
```
OPENVINO_ASSERT(output_paddings.size() == num_outputs || output_paddings.empty());
OPENVINO_ASSERT(output_data_types.size() == num_outputs || output_data_types.empty());
if (output_paddings.emty())
    output_paddings.resize(num_outputs, padding());
if (output_data_types.empty())
    output_data_types.resize(num_outputs, optional_data_type());
```

Reply:
This requires some more changes as in many places we have single element type during primitive creation which is updated later. I suggest changing it separately